lib/net.js

```diff
@@ -664,7 +644,13 @@ function onread(nread, buffer) {
 
   if (self.readableLength === 0) {
     self.readable = false;
```

Comment:
Sorry to pester but this should be no longer needed now. `readableLength` is `0` so `'end'` will be emitted on the next tick and it will already set `readable` to [`false`](https://github.com/nodejs/node/blob/8ccd320549efdbab3608ddcbefca62bd69f8b879/lib/_stream_readable.js#L1019).

It's also possible that `'end'` event will never be emitted, for example if `onread()` returns an error before this point, so I wonder if it's better to only add the `'end'` listener here instead of the constructor.

Reply:
> this should be no longer needed now

Hm, I could have sworn there were test failures when I tired that… anyway, it seems to pass now, so: Done!

> It's also possible that the `'end'` event will never be emitted, for example if `onread()` returns an error before reaching this point, so I wonder if it's better to only add the `'end'` listener here instead of the constructor.

That’s not really the common case, is it? I don’t think I have a strong preference.